index.d.ts

```diff
@@ -0,0 +1,18 @@
+import { cleanup, act, RenderOptions, RenderResult } from 'react-testing-library'
+
+export function renderHook<P, R>(
+  callback: (...args: [P]) => R,
+  options?: {
+    initialProps?: P
+  } & RenderOptions
+): {
+  readonly result: {
+    current: R
+  }
+  readonly unmount: RenderResult['unmount']
+  readonly rerender: (hookProps?: P) => void
```

Comment:
Feels unnecessary to have undefined here. If the `P` is inferred as `undefined`, it should be ok. Otherwise, it should be required to pass the props if there were some initially to avoid surprising results.

Reply:
Looking at the implementation of the `rerender` function, it seems that `hookProps.current`(= `initialProps`) is assigned as a default value.
https://github.com/mpeyper/react-hooks-testing-library/blob/f7cd61035d8fb08cbfac376a5a8da1780708a64e/src/index.js#L26

As long as I see this, we can call `rerender` function with no argument or `initialProps`.

Reply:
I see, so it will use initialProps if none are passed. Makes sense I guess although it feels less explicit and might be confusing for tests.

Reply:
Correct.  Subsequent rerenders will use the previously supplied props until new props are supplied. 

The idea is that the props don't change until the hook is rendered with new props, so hooks that use a conditional array (i.e. `useEffect`, `useLayoutEffect`, `useMemo` and `useCallback`) can be reliably tested.